src/H5Cimage.c

```diff
@@ -654,6 +654,11 @@ H5C__load_cache_image(H5F_t *f)
     } /* end if */
 
 done:
+    if (ret_value < 0) {
+        if (H5_addr_defined(cache_ptr->image_addr))
+            cache_ptr->image_buffer = H5MM_xfree(cache_ptr->image_buffer);
+    }
```

Comment:
Memory leak exposed by these file create failure changes. Free cache image pointer if this function fails.